test/controllers/api/v2/auth_source_externals_controller_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class Api::V2::AuthSourceExternalsControllerTest < ActionController::TestCase
+  valid_attrs = { :name => 'external' }
+
+  test "should get index" do
+    get :index, { }
+    assert_response :success
+    assert_not_nil assigns(:auth_source_externals)
+    auth_source_externals = ActiveSupport::JSON.decode(@response.body)
+    refute auth_source_externals.empty?
+  end
+
+  test "should show auth_source_external" do
+    get :show, { :id => auth_sources(:external).to_param }
+    assert_response :success
+    show_response = ActiveSupport::JSON.decode(@response.body)
+    assert !show_response.empty?
+  end
+
+  test "should update auth_source_external" do
+    put :update, { :id => auth_sources(:external).to_param, :auth_source_external => valid_attrs }
+    assert_response :success
+  end
+
+  test 'allow access to auth source external objects' do
+    setup_user('view', 'authenticators')
+    auth_sources(:external).organizations = User.current.organizations
+    auth_sources(:external).locations = User.current.locations
+    get :show, { :id => auth_sources(:external).to_param }
+    assert_response :success
+  end
+
+  test 'taxonomies can be set' do
+    put :update, { :id => auth_sources(:external).to_param,
+                   :organization_names => [taxonomies(:organization1).name],
+                   :location_ids => [taxonomies(:location1).id] }
+    show_response = ActiveSupport::JSON.decode(@response.body)
+    assert_response :success
+    assert_equal taxonomies(:location1).id, show_response['locations'].first['id']
```

Comment:
I suppose this test fails because of strong params, I think attributes (except for id) must be wrapped in the resource hash, e.g.
```
 put :update, { :id => auth_sources(:external).to_param, :auth_source_external => {
                  :organization_names => [taxonomies(:organization1).name],
                  :location_ids => [taxonomies(:location1).id]
                 }
               }
```